src/main/java/cascading/scheme/hcatalog/SchemaUtils.java

```diff
@@ -0,0 +1,152 @@
+/*
+* Copyright (c) 2007-2016 Concurrent, Inc. All Rights Reserved.
+*
+* Project and contact information: http://www.cascading.org/
+*
+* This file is part of the Cascading project.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package cascading.scheme.hcatalog;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.hive.hcatalog.data.schema.HCatFieldSchema;
+import org.apache.hive.hcatalog.data.schema.HCatSchema;
+
+import cascading.tuple.Fields;
+
+/**
+ * A set of static convenient methods for working with {@link HCatSchema}.
+ */
+final class SchemaUtils {
+
+  private SchemaUtils() {
+  }
+
+  /**
+   * Returns a {@link Set} of lower-case column names from the given {@link Fields}.
+   * <p>
+   * Column names are case-insensitive unique {@link String strings}.
+   * </p>
+   *
+   * @param fields Cascading {@linkplain Fields fields} that should map to column names.
+   * @throws {@code IllegalArgumentException} If {@code fields} contains ignore-case duplicates.
+   */
+  public static Set<String> getLowerCaseFieldNames(Fields fields) {
+    Set<String> names = new HashSet<>(fields.size());
+    Set<String> duplicates = new HashSet<>(fields.size());
+    for (Comparable<?> name : fields) {
+      String lowerCaseName = name.toString().toLowerCase();
```

Comment:
When getting or setting anything in `DefaultHCatRecord` using the `HCatSchema`, everything goes through the [`normalizeName`](https://github.com/apache/hive/blob/master/hcatalog/core/src/main/java/org/apache/hive/hcatalog/data/schema/HCatSchema.java#L144) method, which also does a `toLowerCase` without the `Locale`.